app/javascript/dashboard/composables/useKeyboardEvents.js

```diff
@@ -0,0 +1,79 @@
+import { onMounted, onBeforeUnmount } from 'vue';
+import {
+  isActiveElementTypeable,
+  isEscape,
+} from 'shared/helpers/KeyboardHelpers';
+import { createKeybindingsHandler } from 'tinykeys';
+
+export function useKeyboardEvents(elRef, keyboardEvents) {
+  // this is a store that stores the handler globally, and only gets reset on reload
+  const taggedHandlers = [];
+
+  const getKeyboardEvents = () => {
+    // Define your keyboard events here
+    return keyboardEvents || null;
+  };
+
+  const addEventHandler = keydownHandler => {
+    const indexToAppend = taggedHandlers.push(keydownHandler) - 1;
+    const root = elRef.value;
+    console.log('rootel', root);
+    if (root && root.dataset) {
+      // For the components with a top level v-if Vue renders it as an empty comment in the DOM
+      // so we need to check if the root element has a dataset property to ensure it is a valid element
+      document.addEventListener('keydown', keydownHandler);
+      root.dataset.keydownHandlerIndex = indexToAppend;
+    }
+  };
```

Comment:
A few changes we can do to improve this

1. We can rethink this now to use a `WeakMap` instead of relying on `dataset` here, 
2. The check for root.dataset can be done way before in `onMounted` and we can make this function simpler. 
3. The check can be `root instanceof Element` instead of checking for `dataset`